// angular-tradingview-widget/esm2020/lib/tradingview-widget.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TradingviewWidgetComponent } from './tradingview-widget.component';
import * as i0 from "@angular/core";
export class TradingviewWidgetModule {
}
TradingviewWidgetModule.ɵfac = function TradingviewWidgetModule_Factory(t) { return new (t || TradingviewWidgetModule)(); };
TradingviewWidgetModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: TradingviewWidgetModule });
TradingviewWidgetModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(TradingviewWidgetModule, [{
        type: NgModule,
        args: [{
                declarations: [TradingviewWidgetComponent],
                imports: [
                    CommonModule
                ],
                exports: [TradingviewWidgetComponent],
                providers: []
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(TradingviewWidgetModule, { declarations: [TradingviewWidgetComponent], imports: [CommonModule], exports: [TradingviewWidgetComponent] }); })();
